security/design/oauth2.go

```diff
@@ -0,0 +1,50 @@
+package design
+
+import (
+	. "github.com/goadesign/goa/design"
+	. "github.com/goadesign/goa/design/apidsl"
+)
+
+// OAuth defines a security scheme using OAuth2. The scheme uses the standard "access code" flow
+// where the application must first retrieve an authorization code to request for the refresn and
+// access tokens.
+//
+var OAuth2 = OAuth2Security("oauth2", func() {
+	// /oauth2/code is the path to the action that produces access codes
+	// /oauth2/token is the path to the action that produces refresh and access tokens from
+	// access codes and existing refresh tokens.
+	AccessCodeFlow("/oauth2/code", "/oauth2/token")
+	// All secured API requests must carry the "api" scope.
+	Scope("api")
```

Comment:
It would be best to describe those scopes, as they're added to the documentation in the Swagger file, and explain to the consumer what those scopes mean.